os/pm/pm_idle.c

```diff
@@ -98,6 +98,16 @@ void pm_idle(void)
 		newstate = pm_checkstate();
 #ifdef CONFIG_PM_METRICS
 		pm_metrics_update_idle();
+#endif
+#ifdef CONFIG_PM_TIMEDWAKEUP
+		/* get wakeup timer */
+		if (newstate == PM_SLEEP) {
```

Comment:
is the placement/flow of the PM_TIMEDWAKEUP code matters to resolve the this hang case? If yes then what is the reason?

Reply:
Yes, this placement will reduce the probability of `exiting out` from `pm_idle`. I believe the frequent `exit` cases should be kept at top to avoid going/recovering from unwanted state.